cmd/argocd-util/commands/globalproject.go

```diff
@@ -0,0 +1,148 @@
+package commands
+
+import (
+	"bufio"
+	"io"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/ghodss/yaml"
+	"github.com/spf13/cobra"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/argoproj/gitops-engine/pkg/utils/errors"
+
+	"github.com/argoproj/argo-cd/pkg/apis/application/v1alpha1"
+	"github.com/argoproj/argo-cd/util/cli"
+
+	// load the gcp plugin (required to authenticate against GKE clusters).
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	// load the oidc plugin (required to authenticate with OpenID Connect).
+	_ "k8s.io/client-go/plugin/pkg/client/auth/oidc"
+	// load the azure plugin (required to authenticate with AKS clusters).
+	_ "k8s.io/client-go/plugin/pkg/client/auth/azure"
+)
+
+// NewGlobalProjectGenCommand generates a project from clusterRole
+func NewGlobalProjectGenCommand() *cobra.Command {
+	var (
+		clientConfig clientcmd.ClientConfig
+		out          string
+	)
+	var command = &cobra.Command{
+		Use:   "globalproject CLUSTERROLE_PATH",
+		Short: "Generates global project for the specified clusterRole",
```

Comment:
I think there is no need to inject any name and placeholder values for sources and destinations. Instead, I would suggest renaming command to `argocd projects generate-allow-list <PROJ_NAME>` and output only metadata and spec with namespaceResourceWhitelist e.g. 

```
apiVersion: argoproj.io/v1alpha1
kind: AppProject
metadata:
  name: my-project
spec:
  namespaceResourceWhitelist:
    ...
```

This file can be used as a kustomize patch and insert `namespaceResourceWhitelist` into existing project. This is more convenient then generate a whole project and then manually remove placeholder fields.